// gcommands/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageSelectMenu = exports.MessageButton = exports.MessageActionRow = exports.MessageEmbed = exports.Inhibitor = exports.HandlerManager = exports.Handlers = exports.ListenerManager = exports.Listeners = exports.ComponentManager = exports.Components = exports.CommandManager = exports.Commands = exports.PluginManager = exports.Plugins = void 0;
const tslib_1 = require("tslib");
// Listeners
require("./listeners/Ready");
require("./listeners/InteractionCommandHandler");
require("./listeners/MessageCommandHandler");
require("./listeners/ComponentHandler");
require("./listeners/AutocompleteHandler");
// Client
(0, tslib_1.__exportStar)(require("./lib/GClient"), exports);
var discord_js_1 = require("discord.js");
// Structures
(0, tslib_1.__exportStar)(require("./lib/structures/Plugin"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/Listener"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/Command"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/Component"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/contexts/Context"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/contexts/CommandContext"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/contexts/ComponentContext"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/contexts/AutocompleteContext"), exports);
(0, tslib_1.__exportStar)(require("./lib/structures/Argument"), exports);
// Managers
var PluginManager_1 = require("./lib/managers/PluginManager");
Object.defineProperty(exports, "Plugins", { enumerable: true, get: function () { return PluginManager_1.Plugins; } });
Object.defineProperty(exports, "PluginManager", { enumerable: true, get: function () { return PluginManager_1.PluginManager; } });
var CommandManager_1 = require("./lib/managers/CommandManager");
Object.defineProperty(exports, "Commands", { enumerable: true, get: function () { return CommandManager_1.Commands; } });
Object.defineProperty(exports, "CommandManager", { enumerable: true, get: function () { return CommandManager_1.CommandManager; } });
var ComponentManager_1 = require("./lib/managers/ComponentManager");
Object.defineProperty(exports, "Components", { enumerable: true, get: function () { return ComponentManager_1.Components; } });
Object.defineProperty(exports, "ComponentManager", { enumerable: true, get: function () { return ComponentManager_1.ComponentManager; } });
var ListenerManager_1 = require("./lib/managers/ListenerManager");
Object.defineProperty(exports, "Listeners", { enumerable: true, get: function () { return ListenerManager_1.Listeners; } });
Object.defineProperty(exports, "ListenerManager", { enumerable: true, get: function () { return ListenerManager_1.ListenerManager; } });
var HandlerManager_1 = require("./lib/managers/HandlerManager");
Object.defineProperty(exports, "Handlers", { enumerable: true, get: function () { return HandlerManager_1.Handlers; } });
Object.defineProperty(exports, "HandlerManager", { enumerable: true, get: function () { return HandlerManager_1.HandlerManager; } });
// Logger
(0, tslib_1.__exportStar)(require("./lib/util/logger/Logger"), exports);
// Inhibitors
exports.Inhibitor = (0, tslib_1.__importStar)(require("./inhibitors"));
/* Providers
 * Providers will not be exported due to additional modules. To import, just use the path gcommands/dist/providers/{name}
 */
(0, tslib_1.__exportStar)(require("./lib/structures/Provider"), exports);
// Util
(0, tslib_1.__exportStar)(require("./util/customId"), exports);
(0, tslib_1.__exportStar)(require("./lib/util/registerDirectory"), exports);
(0, tslib_1.__exportStar)(require("./lib/util/registerDirectories"), exports);
(0, tslib_1.__exportStar)(require("./util/confirm"), exports);
(0, tslib_1.__exportStar)(require("./lib/util/Util"), exports);
// Re-exports
var discord_js_2 = require("discord.js");
Object.defineProperty(exports, "MessageEmbed", { enumerable: true, get: function () { return discord_js_2.MessageEmbed; } });
Object.defineProperty(exports, "MessageActionRow", { enumerable: true, get: function () { return discord_js_2.MessageActionRow; } });
Object.defineProperty(exports, "MessageButton", { enumerable: true, get: function () { return discord_js_2.MessageButton; } });
Object.defineProperty(exports, "MessageSelectMenu", { enumerable: true, get: function () { return discord_js_2.MessageSelectMenu; } });
