// gcommands/dist/lib/managers/ComponentManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Components = exports.ComponentManager = void 0;
const discord_js_1 = require("discord.js");
const Component_1 = require("../structures/Component");
const Logger_1 = require("../util/logger/Logger");
const PluginManager_1 = require("./PluginManager");
class ComponentManager extends discord_js_1.Collection {
    register(component) {
        if (component instanceof Component_1.Component) {
            if (this.has(component.name) && !this.get(component.name)?.reloading)
                Logger_1.Logger.warn('Overwriting component', component.name);
            if (this.client)
                component.initialize(this.client);
            if (PluginManager_1.Plugins.currentlyLoading)
                component.owner = PluginManager_1.Plugins.currentlyLoading;
            this.set(component.name, component);
            Logger_1.Logger.emit(Logger_1.Events.COMPONENT_REGISTERED, component);
            Logger_1.Logger.debug('Registered component', component.name, component.owner ? `(by plugin ${component.owner})` : '');
        }
        else
            Logger_1.Logger.warn('Component must be a instance of Component');
        return this;
    }
    unregister(componentName) {
        const component = this.get(componentName);
        if (component) {
            this.delete(componentName);
            Logger_1.Logger.emit(Logger_1.Events.COMPONENT_UNREGISTERED, component);
            Logger_1.Logger.debug('Unregistered component', component.name);
        }
        return component;
    }
    async initiate(client) {
        this.client = client;
        this.forEach(component => component.initialize(client));
    }
}
exports.ComponentManager = ComponentManager;
exports.Components = new ComponentManager();
